include/albatross/src/core/parameter_handling_mixin.hpp

```diff
@@ -220,7 +220,7 @@ class ParameterHandlingMixin {
     if (!map_contains(current_params, key)) {
       std::cerr << "Error: Key `" << key << "` not found in parameters: "
                 << pretty_params(current_params);
-      exit(EXIT_FAILURE);
+      assert(false);
```

Comment:
By default asserts will become no-ops in release mode, so this won't reliably kill the process. An ugly hack we've used elsewhere is to use the cmake `target_compile_definitions` command to undefine the `NDEBUG` preprocessor macro for a particular target. That's difficult to do with a header only library like albatross, so you should double check the downstream uses of albatross for that hack.